Master Sizing (#955)
giantswarm/docs: src/content/kubernetes/best-practices/index.md

For production usage the cluster should have at least 5 nodes and a buffer of 2 nodes.

#### Master nodes

> Master nodes matter much more than worker nodes in order to achieve better availability and performance. Several worker nodes can be replaced at the same time with no downtime but the failure of a master node could cause a major issue if it is your only master node.

Reviewer: Looks like it would make sense to mention [HA masters](https://docs.giantswarm.io/advanced/high-availability/masters/) from this paragraph.

Author: Added HA info

> The master hosts the critical components like the API, scheduler, ETCD... The load of this components will directly depend on the amount of resources they are managing, API requests and events happening in the cluster.

Reviewer: ```suggestion
The master nodes host the critical components like the API, scheduler, Etcd, and many more. The load of these components will directly depend on the amount of resources they are managing, the number of API requests being served, and the events being generated in the cluster.
```

In order to have high availability we recommend:

- Cluster with 50 nodes or 1000 pods should and single master should use m5.2xlarge instance types or higher.

> - Cluster with 100 nodes or 2000 pods should use m5.4xlarge or High Availability with m5.2xlarge instances.

Reviewer: These recommendations are specific to AWS. Since the entire page is not provider-specific, I would like to

- highlight that these recommendations are AWS-specific
- add the equivalent Azure-specific recommendations

Author: @giantswarm/team-celestial do you have a size recommendation for the control plane nodes?

Reviewer 2: Standard_D4s_v3 for masters, Standard_D8s_v3 for workers

In order to have high availability we recommend:

> - Cluster with 50 nodes or 1000 pods should use m5.2xlarge or Standard_D4s_v3 instance types or higher.

Reviewer 2: ```suggestion
- Cluster with 50 nodes or 1000 pods should use m5.2xlarge (on AWS) or Standard_D4s_v3 (on Azure) instance types or higher.
```

> - Cluster with 100 nodes or 2000 pods should use m5.4xlarge or Standard_D8s_v3 in single master or High Availability with m5.2xlarge or Standard_D4s_v3 instances.

Reviewer 2: ```suggestion
- Cluster with 100 nodes or 2000 pods should use m5.4xlarge (on AWS) or Standard_D8s_v3 (on Azure) in single master or High Availability with m5.2xlarge or Standard_D4s_v3 instances.
```
